lib/mongoid/inspectable.rb

```diff
@@ -19,6 +19,38 @@ def inspect
       "#<#{self.class.name} _id: #{_id}, #{inspection * ', '}>"
     end
 
+    # This pretty prints the same information as the inspect method. This is
+    # meant to be called by the default 'pp' gem.
+    #
+    # @param [ PP ] pretty_printer The pretty printer.
+    #
+    # @example Pretty print the document.
+    #   person.pretty_inspect
+    #
+    # @api private
+    def pretty_print(pretty_printer)
+      keys = fields.keys | attributes.keys
+      pretty_printer.group(1, "#<#{self.class.name}", '>') do
+        sep = lambda { pretty_printer.text(',') }
+        pretty_printer.seplist(keys, sep) do |key|
+          pretty_printer.breakable
+          as = if (field = fields[key])
+            field.options[:as]
+          end
+          pretty_printer.text("#{key}#{as ? "(#{as})" : nil}")
```

Comment:
Maybe something like this?

```suggestion
          as = "(#{fields[key].options[:as]})" if fields[key]
          pretty_printer.text("#{key}#{as}")
```

Mostly, the assignment to a trivial `if` statement feels awkward to me; I think it feels more natural to use the `if` statement modifier in that case, even though you wind up duplicating the `fields[key]` query.

Reply:
This doesn't quite work because `fields[key]` can be nil. I created a variation of your suggestion.